### PR TITLE
fix #459: use for-loop instead of Promise.all in migration

### LINKED ISSUE #459: OData filter: date/time fields more precise than millisecond
We are seeing cases where filtering OData does not exclude all the submissions it's expected to. @lognaturel observed while using the `gt` operator:

> If my last update timestamp is `2022-02-23T23:28:50.058Z`, it includes a submission made at timestamp `2022-02-23T23:28:50.058Z`. If I use `2022-02-23T23:28:50.059Z` in the query, that submission is excluded.

I think this is because Postgres timestamp columns are more precise than millisecond. This also came up in #358 (which led to b15086db66af98c39d7d9dd2a9a505c27760421c).

This comes up when fetching any submissions created since the latest known submission. One way to handle this is to ignore duplicate submissions. If that's not possible, 1ms could be added to the latest timestamp.

We could attempt to fix this in the code, probably by truncating any timestamp column. We could truncate a timestamp column in queries in which it is used to filter, though we would probably have to add an index for that. We could also consider truncating the timestamp when it is stored. It looks like the precision of Postgres timestamp columns can be configured (as part of the data type of the column: [docs](https://www.postgresql.org/docs/9.6/datatype-datetime.html)).

Alternatively, perhaps we don't change the code and simply document the issue in the API docs, along with the different strategies to handle it.

### PR DESCRIPTION
**Pull request by sadiqkhoja**
Issue: `20221208-01-reduce-tz-precision.js` is timing out if there is too much data in DB.
Change: Replaced `Promise.all` with for-loop in migration file

### CODE REVIEW

**Comment by matthew-white** [comment]
@alxndrsn, do you have time to take a look at this or try it out locally? It'd be great to resolve this before we deploy to the QA server.

**Comment by matthew-white** [comment]
I'm going to be updating the QA server tomorrow, so let's try to get this merged within the next 24 hours or so so that we can exercise the migration that we intend to deploy. @alxndrsn, the previous version of this migration timed out for you, right? Have you been able to try out this new version? Or @sadiqkhoja, have you been able to validate this new version of the migration another way?

File: lib/model/migrations/20221208-01-reduce-tz-precision.js [hunk 1]
```diff
@@ -7,40 +7,45 @@
...
 // Issue:        #cb459 - `gt` filter for submissionDate is not working as expected because of tz precision
 // Root cause:   Default timestamptz precision in postgres is microseconds and node/js has just milliseconds
 // Solution:     Let's change precision to milliseconds in database, since there is no value in having higher precision
 //               in the database when application can't use/handle it + typical usage of ODK Central doesn't demand higher
 //               precision.
 
-const getTimestampColumns = (db) => db.raw(`
+const getTablesWithTimestampColumns = (db) => db.raw(`
   SELECT
     table_name, column_name, is_nullable
```
Comment (alxndrsn) [inline lib/model/migrations/20221208-01-reduce-tz-precision.js:20]: is `is_nullable` still used?

File: lib/model/migrations/20221208-01-reduce-tz-precision.js [hunk 2]
```diff
@@ -7,40 +7,45 @@
...
   FROM
     information_schema.columns
   WHERE
     table_schema = 'public'
     AND udt_name = 'timestamptz'`)
-  .then(data => data.rows);
+  .then(data => groupBy(r => r.table_name, data.rows));
```
Comment (alxndrsn) [inline lib/model/migrations/20221208-01-reduce-tz-precision.js:26]: Is `groupBy()` equivalent to `GROUP BY`?

e.g.

```sql
SELECT table_name, JSON_AGG(column_name) 
  FROM information_schema.columns
  WHERE table_schema = 'public'
    AND udt_name = 'timestamptz'
  GROUP BY table_name
```

**Review by alxndrsn — APPROVED** [review]
The code looks fine to me, but I no longer have a database in a known-broken state to test this against.

**Pull request merged**